handler/src/main/java/io/netty/handler/ssl/SniHandler.java

```diff
@@ -362,51 +132,6 @@ protected void replaceHandler(ChannelHandlerContext ctx, String hostname, SslCon
         }
```

Comment:
have you considered moving `replaceHandler` into `AbstractSniHandler`? It seems like this may be useful, or some portion of it (to ensure no leak occurs).